core/share.go

```diff
@@ -0,0 +1,57 @@
+package core
+
+import (
+	"context"
+
+	"github.com/deluan/rest"
+	gonanoid "github.com/matoous/go-nanoid"
+	"github.com/navidrome/navidrome/model"
+)
+
+type Share interface {
+	NewRepository(ctx context.Context) rest.Repository
+}
+
+func NewShare(ds model.DataStore) Share {
+	return &shareService{
+		ds: ds,
+	}
+}
+
+type shareService struct {
+	ds model.DataStore
+}
+
+func (s *shareService) NewRepository(ctx context.Context) rest.Repository {
+	repo := s.ds.Share(ctx)
+	wrapper := &shareRepositoryWrapper{
+		ShareRepository: repo,
+		Repository:      repo.(rest.Repository),
+		Persistable:     repo.(rest.Persistable),
+	}
+	return wrapper
+}
+
+type shareRepositoryWrapper struct {
+	model.ShareRepository
+	rest.Repository
+	rest.Persistable
+}
+
+func (r *shareRepositoryWrapper) Save(entity interface{}) (string, error) {
+	s := entity.(*model.Share)
+	id, err := gonanoid.Generate("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz", 9)
+	s.Name = id
+	if err != nil {
```

Comment:
Sorry for the post-merge review. In Go we usually check the result of the error as soon as we have on to check. So in this case, this would be a better way to handle it:

```golang
        id, err := gonanoid.Generate(...)
	if err != nil {
		return "", err
	}
        s.Name = id
```

No need to change, I'm adding tests and will fix it